src/main/settings.ts

```diff
@@ -287,7 +292,40 @@ export class Settings<T extends object> extends PropertyChangeEmitter<T> {
    */
   private saveSettings(updatedSettings: T) {
     this.watcher?.unwatch(this.filePath);
-    fs.writeJSONSync(this.filePath, updatedSettings, { spaces: 2 });
+    try {
+      fs.writeJSONSync(this.filePath, updatedSettings, { spaces: 2 });
+    } catch (error) {
+      // Handle read-only config files correctly.
+      if (error.code === 'EROFS' || error.code === 'EACCES') {
+        // Generate a temporary directory to write the files to for easy reference and write to it.
+        const tmpBaseDir = os.tmpdir() + '/kando/';
```

Comment:
I am not sure if this would work on Windows. I think this should be better:

```suggestion
        const tmpBaseDir = path.join(os.tmpdir(), 'kando');
```